ESP3Parser.js

```diff
@@ -2,160 +2,112 @@
-	}
-
-	/**
-	 * Fills data.
-	 *
-	 * @param byte
-	 */
-	function fillData(byte) {
-		if (tmp.dataOffset <  currentESP3Packet.header.dataLength -1) {
-			currentESP3Packet.data.fill(byte, tmp.dataOffset);
-			tmp.dataOffset++;
-			return;
+	fillData(byte) {
+			if (this.tmp.dataOffset <  this.currentESP3Packet.header.dataLength -1) {
+				this.currentESP3Packet.data.fill(byte, this.tmp.dataOffset);
+				this.tmp.dataOffset++;
+				return;
+			}
+			this.currentESP3Packet.data.fill(byte, this.tmp.dataOffset);
+
+			if(this.currentESP3Packet.header.optionalLength > 0) {
+				this.callbackForNextByte = this.fillOptionalData;
+				return;
+	 		}
+			this.callbackForNextByte = this.fetchCRC8ForDataAndCheck;
 		}
-		currentESP3Packet.data.fill(byte, tmp.dataOffset);
-
-		if(currentESP3Packet.header.optionalLength > 0) {
-			callbackForNextByte = fillOptionalData;
-			return;
- 		}
-		callbackForNextByte = fetchCRC8ForDataAndCheck;
-	}
-
-	/**
-	 * Fills optional data.
-	 *
-	 * @param byte
-	 */
-	function fillOptionalData(byte) {
-		if (tmp.optionalDataOffset <  currentESP3Packet.header.optionalLength -1) {
-			currentESP3Packet.optionalData.fill(byte, tmp.optionalDataOffset);
-			tmp.optionalDataOffset++;
-			return;
+		fillOptionalData(byte) {
+			if (this.tmp.optionalDataOffset <  this.currentESP3Packet.header.optionalLength -1) {
+				this.currentESP3Packet.optionalData.fill(byte, this.tmp.optionalDataOffset);
+				this.tmp.optionalDataOffset++;
+				return;
+			}
+			this.currentESP3Packet.optionalData.fill(byte, this.tmp.optionalDataOffset);
+			this.callbackForNextByte = this.fetchCRC8ForDataAndCheck;
 		}
-		currentESP3Packet.optionalData.fill(byte, tmp.optionalDataOffset);
-		callbackForNextByte = fetchCRC8ForDataAndCheck;
-	}
-
-	/**
-	 * Checks CRC8 for datas(Data + OptionalData) and emits ESP3 Packet or start from scratch if it fails.
-	 *
-	 * @param byte
-	 */
-	function fetchCRC8ForDataAndCheck(byte){
-		callbackForNextByte = waitForSyncByte;
+		fetchCRC8ForDataAndCheck(byte){
+		this.callbackForNextByte = this.waitForSyncByte;
 		var datas = Buffer.concat(
-			[currentESP3Packet.data, currentESP3Packet.optionalData],
-			(currentESP3Packet.header.dataLength + currentESP3Packet.header.optionalLength)
+			[this.currentESP3Packet.data, this.currentESP3Packet.optionalData],
+			(this.currentESP3Packet.header.dataLength + this.currentESP3Packet.header.optionalLength)
 		);
-		if (getCrc8(datas) != byte) {
+		if (this.getCrc8(datas) != byte) {
 			// todo : Log this fail
 			return;
 		}
-		currentESP3Packet.crc8Data = byte;
-		emitFetchedESP3Packet();
+		this.currentESP3Packet.crc8Data = byte;
+		this.emitFetchedESP3Packet();
 	}
-
-	/**
-	 * Emits fetched ESP3 packet on data event.
-	 *
-	 */
-	function emitFetchedESP3Packet() {
-		var out = currentESP3Packet;
-		currentESP3Packet = new ESP3Packet();
-		emitter.emit("data", out);
+	emitFetchedESP3Packet() {
+		var out = this.currentESP3Packet.getRawBuffer();
```

Comment:
Suggestion 1 : Keep "Object Mode"  
```suggestion
		var out = this.currentESP3Packet;
```